db/transaction/transaction_test.go

```diff
@@ -50,4 +57,131 @@ var _ = Describe("Transaction", func() {
+
+			schemaPath := "../../tests/test_schema.yaml"
+			Expect(manager.LoadSchemaFromFile(schemaPath)).To(Succeed())
+			netSchema, _ = manager.Schema("network")
+
+			count = 0
+			transx.EXPECT().ListContext(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).DoAndReturn(func(context.Context, *schema.Schema, transaction.Filter, *transaction.ViewOptions, *pagination.Paginator) (list []*schema.Resource, total uint64, err error) {
+				count++
+				return []*schema.Resource{}, 0, nil
+			}).AnyTimes()
+			transx.EXPECT().CreateContext(gomock.Any(), gomock.Any()).AnyTimes()
+			transx.EXPECT().UpdateContext(gomock.Any(), gomock.Any()).AnyTimes()
+			transx.EXPECT().DeleteContext(gomock.Any(), gomock.Any(), gomock.Any()).AnyTimes()
+			transx.EXPECT().QueryContext(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).AnyTimes()
+		})
+		AfterEach(func(){
+			mockCtrl.Finish()
+		})
+
+		It("First list not cached", func() {
+			Expect(count).To(Equal(0))
+			cachedTx.List(netSchema, tx.Filter{}, nil, nil)
+			Expect(count).To(Equal(1))
+		})
+
+		It("List calls get cached", func() {
+			Expect(count).To(Equal(0))
+			cachedTx.List(netSchema, tx.Filter{}, nil, nil)
+			Expect(count).To(Equal(1))
+			cachedTx.List(netSchema, tx.Filter{}, nil, nil)
+			Expect(count).To(Equal(1))
+			cachedTx.LockList(netSchema, tx.Filter{}, nil, nil, schema.LockRelatedResources)
+			Expect(count).To(Equal(1))
+			cachedTx.LockList(netSchema, tx.Filter{}, nil, nil, schema.SkipRelatedResources)
+			Expect(count).To(Equal(1))
+		})
+
+		It("Fetch calls get cached", func() {
+			Expect(count).To(Equal(0))
+			cachedTx.Fetch(netSchema, tx.Filter{}, nil)
+			Expect(count).To(Equal(1))
+			cachedTx.Fetch(netSchema, tx.Filter{}, nil,)
+			Expect(count).To(Equal(1))
+		})
+
+		It("ListCtx calls get cached", func() {
+			Expect(count).To(Equal(0))
+			cachedTx.ListContext(nil, netSchema, tx.Filter{}, nil, nil)
+			Expect(count).To(Equal(1))
+			cachedTx.ListContext(nil, netSchema, tx.Filter{}, nil, nil)
+			Expect(count).To(Equal(1))
+			cachedTx.LockListContext(nil, netSchema, tx.Filter{}, nil, nil, schema.LockRelatedResources)
+			Expect(count).To(Equal(1))
+			cachedTx.LockListContext(nil, netSchema, tx.Filter{}, nil, nil, schema.SkipRelatedResources)
+			Expect(count).To(Equal(1))
+		})
+
+		It("FetchCtx calls get cached", func() {
+			Expect(count).To(Equal(0))
+			cachedTx.FetchContext(nil, netSchema, tx.Filter{}, nil)
+			Expect(count).To(Equal(1))
+			cachedTx.FetchContext(nil, netSchema, tx.Filter{}, nil,)
+			Expect(count).To(Equal(1))
+			cachedTx.LockFetchContext(nil, netSchema, tx.Filter{}, schema.LockRelatedResources, nil)
+			Expect(count).To(Equal(1))
+			cachedTx.LockFetchContext(nil, netSchema, tx.Filter{}, schema.SkipRelatedResources, nil)
+			Expect(count).To(Equal(1))
+		})
+
+		It("Updates / inserts clear cache", func() {
+			Expect(count).To(Equal(0))
+			cachedTx.List(netSchema, tx.Filter{}, nil, nil)
+			Expect(count).To(Equal(1))
```

Comment:
sorry, but I can't follow what this tests is actually testing: please extract subfunctions for each step